script/deploy/DeployManager.sol

```diff
@@ -0,0 +1,146 @@
+
+    function run() external {
+        // TODO: Use vm.readDir to recursively build this?
+        _runDeployFile(new XOGNSetupScript());
+        _runDeployFile(new OgnOgvMigrationScript());
+        _runDeployFile(new XOGNGovernanceScript());
+    }
+
+    function _runDeployFile(BaseMainnetScript deployScript) internal {
+        string memory chainIdStr = Strings.toString(block.chainid);
+        string memory chainIdKey = string(abi.encodePacked(".", chainIdStr));
+
+        string memory contractsKey = string(abi.encodePacked(chainIdKey, ".contracts"));
+        string memory executionsKey = string(abi.encodePacked(chainIdKey, ".executions"));
+
+        string memory deploymentsFilePath = getDeploymentFilePath();
+        string memory fileContents = vm.readFile(deploymentsFilePath);
+
+        /**
+         * Execution History
+         */
+        string memory currentExecutions = "";
+        string[] memory executionKeys = vm.parseJsonKeys(fileContents, executionsKey);
+
+        for (uint256 i = 0; i < executionKeys.length; ++i) {
+            uint256 deployedTimestamp =
+                vm.parseJsonUint(fileContents, string(abi.encodePacked(executionsKey, ".", executionKeys[i])));
+
+            currentExecutions = vm.serializeUint(executionsKey, executionKeys[i], deployedTimestamp);
+            scriptsExecuted[executionKeys[i]] = true;
+        }
+
+        if (scriptsExecuted[deployScript.DEPLOY_NAME()]) {
+            // TODO: Handle any active governance proposal
+            console.log("Skipping already deployed script");
+            return;
+        }
+
+        /**
+         * Pre-deployment
+         */
+        string memory networkDeployments = "";
+        string[] memory existingContracts = vm.parseJsonKeys(fileContents, contractsKey);
+        for (uint256 i = 0; i < existingContracts.length; ++i) {
+            address deployedAddr =
+                vm.parseJsonAddress(fileContents, string(abi.encodePacked(contractsKey, ".", existingContracts[i])));
+
+            networkDeployments = vm.serializeAddress(contractsKey, existingContracts[i], deployedAddr);
+
+            deployedContracts[existingContracts[i]] = deployedAddr;
+
+            deployScript.preloadDeployedContract(existingContracts[i], deployedAddr);
+        }
+
+        // Deployment
+        deployScript.setUp();
+        deployScript.run();
+
+        /**
+         * Post-deployment
+         */
+        BaseMainnetScript.DeployRecord[] memory records = deployScript.getAllDeployRecords();
+
+        for (uint256 i = 0; i < records.length; ++i) {
+            string memory name = records[i].name;
+            address addr = records[i].addr;
+
+            console.log(string(abi.encodePacked("> Recorded Deploy of ", name, " at")), addr);
+            networkDeployments = vm.serializeAddress(contractsKey, name, addr);
+            deployedContracts[name] = addr;
+        }
+
+        vm.writeJson(networkDeployments, deploymentsFilePath, contractsKey);
```

Comment:
I'm still getting a lot of random failure on these writeJson's (about 2/3 of the 50% of tests fail, about 1/6 of the time, all tests succeed)

Reply:
Yeah. It's mostly about writing twice to the same file back to back. Thats why I added a small delay. Will work on a clean fix to do a single write instead